src/NServiceBus.Transport.RabbitMQ/Configuration/AmqpConnectionString.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+
+namespace NServiceBus.Transport.RabbitMQ
+{
+    class AmqpConnectionString
+    {
+        public static Action<RabbitMQTransport> Parse(string connectionString)
```

Comment:
Well, it could take the instance of the transport and set properties on it. I somehow don't like a mutating static method but I guess this is not much better since it returns a thing that does the mutation.